tools/ci_build/github/linux/docker/scripts/install_deps.sh

```diff
@@ -117,6 +117,12 @@ ${PYTHON_EXE} -m pip install -r ${0/%install_deps\.sh/requirements\.txt}
 if [ $DEVICE_TYPE = "gpu" ]; then
   if [[ $INSTALL_DEPS_TRAINING = true ]]; then
     ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/requirements.txt}
+    # pip runs through the requirements.txt twice. First time downloading the packages and running each package's
+    # setup.py and the second time actually installing it.
+    # deepspeed has `import torch` in its setup.py. So having deepspeed as a part of requirements.txt even though torch
+    # package has already been collected results in an error.
+    # Work around is to install deepspeed separately after the requirements.txt is installed through secondary_requirements.txt
+    ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/secondary_requirements.txt}
```

Comment:
A tiny suggestion: Could you change the name of secondary_requirements.txt to "secondary/requirements.txt". So the filename is still requirements.txt, but it is in a different folder. It will be easier for us to track the dependencies for compliance purposes.

Reply:
or deepspeed/requirements.txt

Reply:
I moved ```secondary_requirements.txt``` to ```secondary/requirements.txt```. This way if there are any other requirements that have a similar dependency on the primary ones, it can go here.
Please let me know if ```deepspeed/requirements.txt``` is  a better suited name for this.

Reply:
cool, that's fine